library/alloc/src/raw_vec.rs

```diff
@@ -493,14 +494,12 @@ where
     alloc_guard(new_layout.size())?;
 
     let memory = if let Some((ptr, old_layout)) = current_memory {
-        debug_assert_eq!(old_layout.align(), new_layout.align());
-        unsafe { alloc.grow(ptr, old_layout, new_layout.size()) }
+        unsafe { alloc.grow(ptr, old_layout, new_layout) }
```

Comment:
I think the regression may be here: the compiler doesn't know that `old_layout.align() == new_layout.align()`.

Reply:
Hmm, possible. But to be sure we could also add an `assume()` and make another perf-run

Reply:
As opt-builds are not affected by the regression, let's try it with the debug assertion only.

Reply:
How does a debug assertion help here?

Reply:
I guess it was a little late last night and I made a mistake, I'm sorry. I'll add `intrinsics::assume`, this may give a speed up.